lib/Readers/ELFSection.cpp

```diff
@@ -107,6 +107,21 @@ std::string ELFSection::getELFPermissionsStr(uint32_t permissions) {
   return elfPermStr;
 }
 
+std::string ELFSection::getSectionAnnotations() const {
+  std::ostringstream oss;
+  for (size_t i = 0; i < Annotations.size(); ++i) {
+    oss << Annotations[i];
+    if (i != Annotations.size() - 1)
+      oss << ", ";
+  }
+  return oss.str();
+}
+
+bool ELFSection::hasAnnotations() const { return !Annotations.empty(); }
+
+void ELFSection::addSectionAnnotation(const std::string &Annotation) {
+  Annotations.push_back(Annotation);
```

Comment:
```
// std::vector<std::string> // 16 bytes
// std::string // data + size = 16 bytes

// std::vector<std::string> // 16 bytes + n * 16
// std::string // 16 bytes
```

Can we think about moving the std::vector to std::string as we discussed as a follow-up / after the current milestone?

Reply:
Yes we should do this as a follow up..